app/models/subnet.rb

```diff
@@ -152,12 +152,16 @@ def cidr=(cidr)
     nil
   end
 
-  def supports_ipam_mode?(mode)
-    self.class.supported_ipam_modes.include?(mode)
+  def self.supports_ipam_mode?(mode)
```

Comment:
Nitpick, move all `self.` methods to `class << self` below the validations, this class is becoming a bit unwieldy